Assets/Editor/E07_Data.cs

```diff
@@ -0,0 +1,151 @@
+            }
+
+            this.GetRootVisualContainer().AddChild(scrollView);
+            this.GetRootVisualContainer().AddStyleSheetPath("TreeView");
+        }
+
+        static void UpdateName(VisualElement treeViewItem, Object data)
+        {
+            if (data == null)
+                return;
+
+            treeViewItem.Q<Label>().text = data.name;
+        }
+
+        public void UpdateChildren(VisualElement treeViewItem, Object data)
+        {            
+            Transform transform = data as Transform;           
+
+            // Might be deleted
+            if (transform == null)
+                return;
+
+            List<GameObject> gos = new List<GameObject>();
+            for (int i = 0; i < transform.childCount; i++)
+            {
+                gos.Add(transform.GetChild(i).gameObject);
+            }
+            Toggle toggle = treeViewItem.Q<Toggle>();
+			toggle.enabled = gos.Count > 0;
+			toggle.on = toggle.enabled && m_ExpandedGOs.Contains(transform.gameObject.GetInstanceID());
+
+            if (m_ExpandedGOs.Contains(transform.gameObject.GetInstanceID()))
+            {
+                RecursiveSetUp(gos, treeViewItem.Q<VisualContainer>("ChildrenContainer"));    
+            }
+        }
+
+        public void RecursiveSetUp(IEnumerable<GameObject> gameObjects, VisualContainer container)
+        {
+            var currentGOs = new List<GameObject>();
+            container.ToList().ForEach((e) => {
+                GameObject go = e.data as GameObject;
+
+                // remove gone objects
+                if (go == null || !gameObjects.Contains(go))
+                {
+                    container.RemoveChild(e);
+                }
+                // or-remember those that already have an item
+                else
+                {
+                    currentGOs.Add(go);
+                }
+            });
+
+
+            foreach(GameObject go in gameObjects.Except(currentGOs))
+            {
+                VisualContainer added = asset.CloneTree();
+                added.data = go;
+
+                UpdateName(added, go);
+                added.RegisterWatch(go, UpdateName);
+				UpdateChildren(added, go.transform);
+				added.RegisterWatch(go.transform, UpdateChildren);
+				
+				added.Q<Toggle>().OnToggle(() =>
+				{
+                    OnToggle(added);
+				});
+
+                container.AddChild(added);
+			}
+
+            container.Sort((a, b) => {
+                Transform aTrans = (a.data as GameObject).transform;
+                Transform bTrans = (b.data as GameObject).transform;
+                Debug.Assert(aTrans.parent == bTrans.parent);
+				int order = aTrans.GetSiblingIndex() - bTrans.GetSiblingIndex(); 
+                return order;
+            });
+        }
+
+        void OnToggle(VisualContainer treeViewItem)
+        {
+			GameObject go = treeViewItem.data as GameObject;
+            Debug.Assert(go != null, "No GO attached to item!");
```

Comment:
Let's see how in the next PR we can have a ViewData service to store and retrieve that.

// Off the cuff?
var v = ViewDataStore.GetOrCreateDefault<int>(key);
var v = ViewDataStore.Store<int>(key, 5);